### DISCUSSION TITLE
Adding an ERD Tool to Supabase.

### DISCUSSION THREAD

**Discussion by segheysens** [supabase/supabase#9901] [Feature Requests]
It would be wonderful to have an auto-diagramming tool like [pgAdmin's ERD Tool](https://www.pgadmin.org/docs/pgadmin4/latest/erd_tool.html) built into Supabase, for visualizing the tables in your database. Although I think that pgAdmin allows you to create draft diagrams, I see the most value in a view-only, autogenerated ERD diagram from the existing tables. In the long term, it could be interesting to be able to edit tables/columns via an ERD diagram, but just visualizing the current state would be a great step forward.

This feature would be useful for developers and other team members when planning and discussing changes to their data model.

<img width="525" alt="erd_tool" src="https://user-images.githubusercontent.com/3592134/198298195-8314920d-d43b-4333-ae29-70c8f5ff6525.png">

The pgAdmin repository is open-source, so we would not need to completely reinvent the wheel: for example, here's [the ERDTool component](https://github.com/pgadmin-org/pgadmin4/blob/master/web/pgadmin/tools/erd/static/js/erd_tool/components/ERDTool.jsx).

**Comment by dshukertjr** [comment]
You can find a community built schema visualizer here!
https://supabase-schema.vercel.app/
